### ISSUE TITLE
Failure when undoing snippet expansion

### ISSUE THREAD

**Issue by mMontu** [SirVer/ultisnips#947]
I'm creating a snippet and noticed that despite it works fine when any change is done to the placeholders, it fails when undoing without any changes:

	snippet svi "Set Variable If"
	\${${1:var}}=    Set Variable If    '\${${2:rc}}' != '${3:abc}'
	`!p snip.rv = '...' + 'a' * (len(t[1]) + 23)` ${4:${value true}}
	...                           ${5:${value false}}
	endsnippet

Starting with the text `    svi`, issuing tab followed by `u` results in:

        svi
    ...a...aaaaaaaaaaaaaaaaaaaaaaaa

Just deleting the second lines issues the following error:

    RecursionError: maximum recursion depth exceeded in comparison

I've tried to follow the examples in the documentation, but it has a similar issue:

```
------------------- SNIP -------------------
snippet wow
${1:Text}`!p snip.rv = (75-2*len(t[1]))*' '+t[1].upper()`
endsnippet
------------------- SNAP -------------------
```

Expanding this snippet and undoing it leaves the following:

        wow                                                                    WOW 

(it inserts spaces and `WOW` at the end of the line).

**Comment by andreyorst** [comment]
Can't reproduce it. What filetype is it? I can't deduce it from snippet definition, sorry. I've tryed to create theese snippets in C filetype, and undoing works for me. I'm not sure that I've understand you right, but I expect that you expanding a snippet with tab, then switchingg to normal mode, and pressing u? So the keystrokes for this are <kbd>s</kbd><kbd>v</kbd><kbd>i</kbd><kbd>Tab</kbd><kbd>Esc</kbd><kbd>u</kbd>, right?

**Comment by mMontu** [comment]
@andreyorst it didn't occur to me that the issue could be filetype specific. The filetype is robot, updating [this file](https://github.com/honza/vim-snippets/blob/master/UltiSnips/robot.snippets).

The keystrokes are exactly those you mentioned.

I've checked and couldn't reproduce the issue with C filetype. Then I went back to robot, and to my surprise none of the snippets I mentioned above failed anymore. I've no idea what have changed -- I didn't update gVim or python, and I have updated Ultisnips before opening this issue.

Anyway, thanks for your help!
